### ISSUE TITLE
Get rid of gnome2 package set

### ISSUE THREAD

**Issue by jtojnar** [NixOS/nixpkgs#39976]
There is `gnome2` package set under desktops but we do not really have GNOME 2 desktop packaged – it just contains various outdated libraries. We should try to remove as much as we can and merge the rest into main namespace.

Generated with the folloeing `git grep -l GConf`

- [x] `nixos/modules/services/x11/window-managers/metacity.nix`
- [ ] `pkgs/applications/audio/google-play-music-desktop-player/default.nix`
- [x] `pkgs/applications/audio/paprefs/default.nix` – https://bugs.freedesktop.org/show_bug.cgi?id=57743
- [x] `pkgs/applications/audio/spotify/default.nix`
- [x] `pkgs/applications/editors/android-studio/common.nix` #322831 
- [x] `pkgs/applications/editors/android-studio/default.nix`
- [x] `pkgs/applications/editors/typora/default.nix`
- [x] `pkgs/applications/graphics/pencil/default.nix`
- [x] `pkgs/applications/misc/airtame/default.nix`
- [ ] `pkgs/applications/misc/gxneur/default.nix` – https://github.com/AndrewCrewKuznetsov/xneur-devel/issues/27
- [x] `pkgs/applications/misc/hyper/default.nix`
- [x] `pkgs/applications/misc/terminus/default.nix`
- [x] `pkgs/applications/misc/tomboy/default.nix`
- [x] `pkgs/applications/networking/browsers/chromium/common.nix`
- [x] `pkgs/applications/networking/browsers/opera/default.nix`
- [x] `pkgs/applications/networking/instant-messengers/discord/default.nix`
- [x] `pkgs/applications/networking/instant-messengers/franz/default.nix`
- [x] `pkgs/applications/networking/instant-messengers/messenger-for-desktop/default.nix`
- [x] `pkgs/applications/networking/instant-messengers/signal-desktop/beta.nix`
- [x] `pkgs/applications/networking/instant-messengers/signal-desktop/default.nix`
- [x] `pkgs/applications/networking/instant-messengers/skypeforlinux/default.nix`
- [x] `pkgs/applications/networking/instant-messengers/slack/default.nix`
- [x] `pkgs/applications/networking/instant-messengers/stride/default.nix`
- [x] `pkgs/applications/networking/instant-messengers/wire-desktop/default.nix`
- [x] `pkgs/applications/networking/mailreaders/inboxer/default.nix`
- [x] `pkgs/applications/networking/mailreaders/nylas-mail-bin/default.nix`
- [x] `pkgs/applications/office/libreoffice/default.nix` - #68414
- [x] `pkgs/applications/office/libreoffice/still.nix`
- [x] `pkgs/applications/office/planner/default.nix`
- [x] `pkgs/applications/version-management/gitkraken/default.nix`
- [x] `pkgs/applications/video/coriander/default.nix` #322986
- [x] `pkgs/applications/video/webtorrent_desktop/default.nix`
- [x] `pkgs/applications/window-managers/xmonad/log-applet/default.nix`
- [x] `pkgs/desktops/gnome-2/bindings/gnome-python/default.nix`
- [ ] `pkgs/desktops/gnome-2/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/gnome-control-center/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/gnome-desktop/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/gnome-keyring/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/gnome-panel/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/gnome-settings-daemon/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/libgnomekbd/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/libgweather/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/mail-notification/default.nix`
- [x] `pkgs/desktops/gnome-2/desktop/metacity/default.nix`
- [ ] `pkgs/desktops/gnome-2/platform/GConf/default.nix`
- [x] `pkgs/desktops/gnome-2/platform/at-spi/default.nix`
- [x] `pkgs/desktops/gnome-2/platform/gconfmm/default.nix`
- [x] `pkgs/desktops/gnome-2/platform/gnome-vfs/default.nix` #322862
- [x] `pkgs/desktops/gnome-2/platform/gtkhtml/4.x.nix`
- [x] `pkgs/desktops/gnome-2/platform/gtkhtml/default.nix`
- [x] `pkgs/desktops/gnome-2/platform/libbonoboui/default.nix` #322862
- [x] `pkgs/desktops/gnome-2/platform/libgnome/default.nix` #322862
- [x] `pkgs/desktops/gnome-2/platform/libgnomeui/default.nix` #322862
- [x] `pkgs/desktops/gnome-3/core/gconf/default.nix`
- [x] `pkgs/desktops/gnome-3/devtools/nemiver/safe_ptr.patch`
- [x] `pkgs/development/compilers/openjdk/10.nix` https://github.com/NixOS/nixpkgs/issues/73246
- [x] `pkgs/development/compilers/openjdk/8.nix` #322599 
- [ ] `pkgs/development/haskell-modules/hackage-packages.nix`
- [x] `pkgs/development/libraries/geoclue/default.nix` – https://github.com/NixOS/nixpkgs/pull/43868
- [ ] `pkgs/development/libraries/gtk-sharp/2.0.nix`
- [x] `pkgs/development/libraries/gtk-sharp/3.0.nix`
- [x] `pkgs/development/libraries/libgksu/default.nix`
- [x] `pkgs/development/libraries/qt-4.x/4.8/default.nix` – https://github.com/NixOS/nixpkgs/issues/33248
- [x] `pkgs/development/libraries/qt-4.x/4.8/dlopen-gtkstyle.diff`
- [x] `pkgs/development/libraries/wxwidgets/wxGTK28.nix`
- [x] `pkgs/development/libraries/wxwidgets/wxGTK29.nix`
- [x] `pkgs/development/libraries/wxwidgets/wxGTK30.nix`
- [x] `pkgs/development/tools/react-native-debugger/default.nix`
- [x] `pkgs/development/tools/unity3d/default.nix`
- [x] `pkgs/development/web/insomnia/default.nix`
- [x] `pkgs/development/web/postman/default.nix`
- [x] `pkgs/games/gtetrinet/default.nix` #322986
- [x] `pkgs/games/steam/fhsenv.nix`
- [x] `pkgs/misc/emulators/mess/default.nix`
- [x] `pkgs/tools/filesystems/relfs/default.nix`
- [ ] `pkgs/tools/misc/alarm-clock-applet/default.nix`
- [x] `pkgs/tools/misc/staruml/default.nix`
- [ ] `pkgs/tools/misc/sweep-visualizer/default.nix`
- [x] `pkgs/tools/security/keybase/gui.nix`
- [ ] `pkgs/top-level/all-packages.nix`

**Comment by jtojnar** [comment]
Shutter is the last package depending on goocanvas 1.0 (via perl module) but GTK 3 support still has long way to go https://launchpad.net/shutter/+milestone/road-to-gtk3. It is an optional dependency, do we keep it?

**Comment by mkg20001** [comment]
This could break some things for cinnamon support as it depends on, among others, gnome2 gconf.

**Comment by jtojnar** [comment]
Wait, what? Is not Cinnamon fork of GNOME 3?

**Comment by mkg20001** [comment]
Don't know if it's all still needed, but found this

```
./cinnamon-common/default.nix:26:, gnome2
./cinnamon-common/default.nix:80:    gnome2.gtkdoc
./cinnamon-session/default.nix:23:, gnome2
./cinnamon-session/default.nix:58:    gnome2.GConf
```

**Comment by jtojnar** [comment]
- `cinnamon-common`: if this is fork of gnome-common, it should not be needed for anything that was maintained in last five years.

- `gnome2.gtkdoc` is an alias of `gtk-doc`. 

- `cinnamon-session`: GConf appears to be optional https://github.com/linuxmint/cinnamon-session/blob/386284e1c7a743c8a8b4162d05fa9da8ed8fe04c/meson_options.txt, only used for GConf-conditional desktop files, which there should not be any. It supports GSettings as well.

**Comment by mkg20001** [comment]
cinnamon-common is actually just a rename of cinnamon (other distros, like ubuntu, seem to rename it as well), since it otherwise clashes with the name of the scope, which also is cinnamon

Resolved the other things on my branch: https://github.com/mkg20001/nixpkgs/tree/mkg-patch

**Comment by mweinelt** [comment]
Relevant for the python2 removal effort (pygtk, pygobject2, etc.).

**Comment by Mindavi** [comment]
I went through the whole list and checked anything that was removed or didn't contain a reference to `{gtk2,GConf}`. Also updated some paths that were still relevant.

**Comment by piegamesde** [comment]
With a few exceptions, yes. Note though that the list is incomplete, especially you are missing out on packages written using Gtk bindings (grep for gtk-sharp, gtkmm, etc. for these).

As much as I'd like to help with this, I'm still pretty burnt out after my (totally incomplete) libglade bulk removal. Therefore I won't do any mass removals until we have the proper infrastructure in place for doing them more easily. (I'm working on that part, but slowly and nothing to publicly show yet)

**Comment by tomodachi94** [comment]
Here are all of the Haskell packages we have which appear to have a direct dependency on gconf (either the Hackage package or the `gnome2` package):

- [ ] gconf
- [ ] gnome-desktop
- [ ] gtkrsync
- [ ] hawitter
- [ ] manatee-core

Generated with `grep "\bgconf\b" -B 6 -A 0 pkgs/development/haskell-modules/hackage-packages.nix | grep ".* = callPackage" | sed 's/"\(.*\)" = callPackage/- [ ] \1/g'`.

**Comment by emilazy** [comment]
It’ll probably be a while before we can remove GTK 2 entirely, but I don’t think we’d want to add more GNOME 2 / GTK 2 dependencies for old versions of software that are [long‐unsupported upstream](https://endoflife.date/unity).
